quickstarts/301-sap-gateway/gateway-principal/main.tf

```diff
@@ -0,0 +1,53 @@
+terraform {
+  required_version = ">= 1.5"
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+    }
+    azuread = {
+      source = "hashicorp/azuread"
+    }
+  }
+}
+
+provider "azuread" {
+  client_id     = var.client_id
```

Comment:
I would keep all examples using CLI unless there is a specific need of using client/secret